### PR TITLE
[FIX] Bug #353 crash when selecting a payment mode that has a variable link to bank accounts

### PR DESCRIPTION
**Pull request by alexis-via**
Cf bug #353 for a detailed scenario of the crash.

I you have better ideas to fix this bug, please propose. I suggests this solution because it's the only one I found that currently works on v10.

### CODE REVIEW

File: account_payment_order/models/account_payment_order.py [hunk 1]
```diff
@@ -162,10 +146,18 @@ def create(self, vals):
     @api.onchange('payment_mode_id')
     def payment_mode_id_change(self):
         journal_id = False
+        res = {'domain': {
+            'journal_id': "[('id', '=', False)]",
```
Comment (pedrobaeza) [inline account_payment_order/models/account_payment_order.py:150]: This should be `{'journal_id': []}`

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by sbidoul** [comment]
@alexis-via I think the onchange will not trigger when opening a previously saved payment order, so in that case, the domain will be wrong.

Dynamic domains are a pain to get right...

I think it's part of the reasons that pushed @lmignon to create https://github.com/OCA/web/pull/567. You may want to try it.

**Comment by alexis-via** [comment]
@sbidoul Very interesting, thanks !
Another interesting bug report https://github.com/odoo/odoo/issues/16072

File: account_payment_order/models/account_payment_order.py (hunk 1 above)
Comment (alexis-via) [reply account_payment_order/models/account_payment_order.py:150]: If I make your modification, then I have access to all journals when no payment mode is selected, which is not what I want : I want the user to select a payement mode first. So I prefer my solution, which works well.

**Comment by alexis-via** [comment]
I propose to merge this to avoid the crash. And then, if someone is able to propose a better solution, he will propose another PR.
I don't want to leave an horrible crash like this one too long in an important OCA module.

**Review by sbidoul — APPROVED** [review]

File: account_payment_order/models/account_payment_order.py (hunk 1 above)
Comment (pedrobaeza) [reply account_payment_order/models/account_payment_order.py:150]: Then please hide the field when no payment mode is selected in the view via `attrs` attribute.

**Pull request merged**